src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -132,76 +132,153 @@ SDCPUThread::advanceInst(TheISA::PCState next_pc)
 
     const InstSeqNum seq_num = lastCommittedInstNum + inflightInsts.size() + 1;
 
-    DPRINTF(SDCPUThreadEvent, "advanceInst(seq %d, pc %#x, upc %#x)\n",
+    DPRINTF(SDCPUThreadEvent, "advanceInst(seq %d, pc %#x, upc %d)\n",
                               seq_num,
                               next_pc.pc(),
                               next_pc.upc());
-    DPRINTF(SDCPUThreadEvent, "Buffer size increased to %d\n",
-                              inflightInsts.size());
+
+    StaticInstPtr static_inst = nullptr;
 
     if (isRomMicroPC(next_pc.microPC())) {
-        const StaticInstPtr static_inst =
+        assert(!curMacroOp);
```

Comment:
This assert is not valid, since we actually use the macroOp to extract things from the microROM.